legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabWorkspaceApi.java

```diff
@@ -310,6 +310,7 @@ public Workspace newWorkspace(String projectId, String workspaceId, WorkspaceTyp
     {
         LegendSDLCServerException.validateNonNull(projectId, "projectId may not be null");
         LegendSDLCServerException.validateNonNull(workspaceId, "workspaceId may not be null");
+        LegendSDLCServerException.validateNonNull(this.getProjectConfiguration(projectId), "Project is not correctly configured");
```

Comment:
This is not the right way to check this. The LegendSDLCServerException.validateX are really for input validation, which is not what this is. This should be validated after the call to validateWorkspaceId. If no project configuration is found, I would recommend throwing an exception with a 409 status and a message explaining that the project structure has not been set up.